### PR TITLE
Feature project library

### PR DESCRIPTION
**Pull request by Belinskiy21**
after review

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
 
-if File.exist? ("project_library.yml")
-	lib.menu
-else
-	lib.put_data
-
-	lib.menu
-end
-
+	if File.exist? ("project_library.yml")
```
Comment (yamnich) [inline main.rb]: please setup your editor first

File: models/library.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  def top_reader
+    top(1, :reader).name
+  end
+
+  def top_book
+    top(1, :book).title
+  end
+
+  def mainstream_readers
+    top(3, :book) do |books|
```
Comment (yamnich) [inline models/library.rb]: it is not very good to use 2 loops

File: models/library.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      @orders.flat_map { |order| order.reader if books.include?(order.book) }
```
Comment (yamnich) [inline models/library.rb]: books already contain orders , no need to select them again from all orders, map and flatten and group by reader and count will be enough

Comment (Belinskiy21) [inline models/library.rb]: books not contain orders(in this place books contain 3 book with author, name and biography information) . orders contain book that is why : @orders.flat_map { |order| order.reader if books.include?(order.book) }
.compact.uniq.size

File: models/library.rb (hunk 2 above)
Comment (Belinskiy21) [inline models/library.rb]: My first decision was not include loops at all. Can you show me, please, what will be acceptable in this case?

File: models/library.rb (hunk 3 above)
Comment (Belinskiy21) [inline models/library.rb]: books doesn't contain orders.

**Review by yamnich — APPROVED** [review]

**Pull request merged**